pkg/compose/port.go

```diff
@@ -44,9 +44,23 @@ func (s *composeService) Port(ctx context.Context, projectName string, service s
 	}
 	container := list[0]
 	for _, p := range container.Ports {
-		if p.PrivatePort == uint16(port) && p.Type == options.Protocol {
+		if p.PrivatePort == port && p.Type == options.Protocol {
 			return p.IP, int(p.PublicPort), nil
 		}
 	}
-	return "", 0, err
+	return "", 0, portNotFoundError(options.Protocol, port, container)
+}
+
+func portNotFoundError(protocol string, port uint16, ctr moby.Container) error {
+	formatPort := func(protocol string, port uint16) string {
+		return fmt.Sprintf("%s/%d", protocol, port)
```

Comment:
Don't these usually get represented as port/protocol (e.g., `80/tcp` or `53/udp`)?
```suggestion
		return fmt.Sprintf("%d/%s", port, protocol)
```

Reply:
Yes, correct; also see https://github.com/docker/cli/pull/3892

Reply:
You know what's sad is that I looked at the convention and tried to copy it and never noticed I had it reversed despite that 🤦